MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/extrusions/RotationExtrusionActivity.java

```diff
@@ -0,0 +1,186 @@
+
+    gyro = sensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER);
+    magnetic = sensorManager.getDefaultSensor(Sensor.TYPE_MAGNETIC_FIELD);
+
+    sensorManager.registerListener(this, gyro , 200);
+    sensorManager.registerListener(this, magnetic , 200);
+  }
+
+  private void setupBuildings() {
+    FillExtrusionLayer fillExtrusionLayer = new FillExtrusionLayer("3d-buildings", "composite");
+    fillExtrusionLayer.setSourceLayer("building");
+    fillExtrusionLayer.setFilter(eq("extrude", "true"));
+    fillExtrusionLayer.setMinZoom(15);
+    fillExtrusionLayer.setProperties(
+            fillExtrusionColor(Color.LTGRAY),
+            fillExtrusionHeight(Function.property("height", new IdentityStops<Float>())),
+            fillExtrusionBase(Function.property("min_height", new IdentityStops<Float>())),
+            fillExtrusionOpacity(0.9f)
+    );
+    mapboxMap.addLayer(fillExtrusionLayer);
+  }
+
+  @Override
+  public boolean onCreateOptionsMenu(Menu menu) {
+    return true;
+  }
+
+  @Override
+  public boolean onOptionsItemSelected(MenuItem item) {
+    return true;
+  }
+
+  @Override
+  protected void onStart() {
+    super.onStart();
+    mapView.onStart();
+  }
+
+  @Override
+  protected void onResume() {
+    super.onResume();
+    mapView.onResume();
+  }
+
+  @Override
+  protected void onPause() {
+    super.onPause();
+    mapView.onPause();
+  }
+
+  @Override
+  protected void onStop() {
+    super.onStop();
+    mapView.onStop();
+  }
+
+  @Override
+  public void onSaveInstanceState(Bundle outState) {
+    super.onSaveInstanceState(outState);
+    mapView.onSaveInstanceState(outState);
+  }
+
+  @Override
+  public void onLowMemory() {
+    super.onLowMemory();
+    mapView.onLowMemory();
+  }
+
+  @Override
+  public void onDestroy() {
+    super.onDestroy();
+    mapView.onDestroy();
+  }
+
+  @Override
+  public void onSensorChanged(SensorEvent event) {
+
+    if (event.sensor.getType() == Sensor.TYPE_ACCELEROMETER) {
+      gravArray = event.values;
+    }
+
+    if (event.sensor.getType() == Sensor.TYPE_MAGNETIC_FIELD) {
+      magneticArray = event.values;
+    }
+
+    if (gravArray != null && magneticArray != null) {
+      float[] num1 = new float[9];
+      float[] num2 = new float[9];
+      boolean success = SensorManager.getRotationMatrix(num1, num2, gravArray, magneticArray);
+      if (success) {
+        float[] orientation = new float[3];
+        SensorManager.getOrientation(num1, orientation);
+        azimut = orientation[0];
+        pitch = orientation[1];
+        roll = orientation[2];
+
+        CameraPosition position = new CameraPosition.Builder()
+                .tilt(pitch * -90)
+                .bearing(roll * 90)
+                .build();
+
+        if (mapboxMap != null) {
+          mapboxMap.animateCamera(CameraUpdateFactory
+                  .newCameraPosition(position), 100
+          );
+        }
+      }
+    }
+
+  }
```

Comment:
above method is 30 lines long, this can be simplified to make the code more readable